mifosng-android/src/main/java/com/mifos/mifosxdroid/SplashScreenActivity.java

```diff
@@ -36,5 +42,44 @@ protected void onCreate(Bundle savedInstanceState) {
             startActivity(intent);
         }
         finish();
+        startappintro(this);
+    }
+    //App intro shred preferences code
+    private  void startappintro(final Context context) {
+        mthread = new Thread(new Runnable() {
+                @Override
+            public void run() {
+                //  Initialize SharedPreferences
+                SharedPreferences PREF = PreferenceManager
+                        .getDefaultSharedPreferences(context);
+
+                //  Create a new boolean and preference and set it to true
+                misFirstStart = PREF.getBoolean("firstStart", true);
+
+                //  If the activity has never started before...
+                if (misFirstStart) {
+
+                    //  Launch app intro
+                    final Intent INTENT = new Intent(SplashScreenActivity.this, AppIntroUser.class);
+
+                    runOnUiThread(new Runnable() {
+                        @Override
+                        public void run() {
+                            startActivity(INTENT);
+                        }
+                    });
+
+                    //  Make a new preferences editor
+                    SharedPreferences.Editor E = PREF.edit();
```

Comment:
Do not do all the Preference related operations inside the activity. Use predefined helper methods from the `PrefManager` class in the `utils` folder.  Give the variables a descriptive name using **Camel Case** naming style